code/modules/power/prototype_shield_emitter.dm

```diff
@@ -0,0 +1,199 @@
+	else
+		to_chat(user, "<span class='warning'>You unbolt the [src] from the floor!</span>")
+	return TRUE
+
+/obj/machinery/power/proto_sh_emitter/attackby(obj/item/I, mob/user, params)
+	if(I.GetID())
+		if(allowed(user))
+			if(is_on)
+				locked = !locked
+				to_chat(user, "<span class='notice'>You [locked ? "lock" : "unlock"] the controls.</span>")
+			else
+				to_chat(user, "<span class='warning'>The controls can only be locked when \the [src] is online!</span>")
+		else
+			to_chat(user, "<span class='danger'>Access denied.</span>")
+		return
+
+/obj/machinery/power/proto_sh_emitter/crowbar_act(mob/living/user, obj/item/I)
+	default_deconstruction_crowbar(I)
+	return TRUE
+
+/obj/machinery/power/proto_sh_emitter/screwdriver_act(mob/living/user, obj/item/I)
+	if(..())
+		return TRUE
+	default_deconstruction_screwdriver(user, "proto_sh_emitter-open", "proto_sh_emitter", I)
+	return TRUE
+
+/obj/machinery/power/proto_sh_emitter/process()
+	if(powered())
+		apc_power = TRUE
+		return
+	if(!powered() && apc_power == TRUE)
+		apc_power = FALSE
+		var/turf/Turf = get_turf(src)
+		is_on = FALSE
+		QDEL_LIST(signs)
+		update_icon_state()
+		message_admins("[src] turned off at [ADMIN_VERBOSEJMP(Turf)]")
+		log_game("[src] turned off at [AREACOORD(Turf)]")
+
+/obj/machinery/power/proto_sh_emitter/interact(mob/user)
+	add_fingerprint(user)
+	if(!anchored)
+		to_chat(user, "<span class='warning'>You need to anchor the [src] first!</span>")
+		return
+	if(!apc_power)
+		to_chat(user, "<span class='warning'>There is no power in this area!!</span>")
+		return
+	if(locked)
+		to_chat(user, "<span class='warning'>The controls are locked!</span>")
+		return
+	if(is_on)
+		remove_barrier(user)
+		return
+	check_map_borders(2,5,2,5)
+	if(!borders)
+		to_chat(user, "<span class='warning'>The motors whir and fail!</span>")
+		return
+	if(normal)
+		build_barrier(1,2,1,4,2,1,2,5,user)
+	else
+		build_barrier(0,2,2,3,1,1,3,4,user)
+
+/** The vars you'll see in the proc() are referred to a mob looking north; NEx NEy refers to the North East corner x and y,
+*all the other vars works in a similar way (N = North, S = South, E = East, W = West x = x axis, y = y axis, i = internal, o = outline). This way of naming the vars
+*won't have much sense for the other directions, so always refer to the north direction when making changes as all other are already properly setup
+*This proc builds the barriers
+**/
+/obj/machinery/power/proto_sh_emitter/proc/build_barrier(SWxi,SWyi,NExi,NEyi,SWxo,SWyo,NExo,NEyo,mob/user)
+	///Stores the outline of the room to generate
+	var/list/outline
+	///Stores the internal turfs of the room to generate
+	var/list/internal
+	var/turf/EmitterTurf = get_turf(src)
+	to_chat(user, "<span class='warning'>You start to turn on the [src] and the generated shields!</span>")
+	if(do_after(user, 1.5 SECONDS, target = src))
+		to_chat(user, "<span class='warning'>You turn on the [src] and the generated shields!</span>")
```

Comment:
```suggestion
	to_chat(user, "<span class='warning'>You start to turn on the [src] and the generated shields!</span>")
	if(!do_after(user, 1.5 SECONDS, target = src))
		return
	to_chat(user, "<span class='warning'>You turn on the [src] and the generated shields!</span>")
	//Stores the outline of the room to generate
	var/list/outline = list()
	//Stores the internal turfs of the room to generate
	var/list/internal = list()
	var/turf/EmitterTurf = get_turf(src)
```
No need to use lazy lists on lists defined inside a proc.
Also, early return and whatnot.